helpers/LinuxUSB.py

```diff
@@ -58,6 +66,21 @@ def AfterConnect(ADBSerialId):
     return 1, SDKVersion, WhatsAppapkPath, versionName, sdPath
 
 
+def DownloadApk(url, fileName):
+    # Streaming, so we can iterate over the response.
+    response = requests.get(url, stream=True)
+    totalSizeInBytes = int(response.headers.get('content-length', 0))
+    blockSize = 1024  # 1 Kibibyte
+    progressBar = tqdm(total=totalSizeInBytes, unit='iB', unit_scale=True)
+    with open(fileName, 'wb') as file:
+        for data in response.iter_content(blockSize):
+            progressBar.update(len(data))
+            file.write(data)
```

Comment:
Instead of directly writing to this file write to a temp file first and rename that temp file to LegacyWhatsApp.apk or whatever this would solve the issue where this try to install incomplete apk and get INSTALL_PARSE_FAILED_NOT_APK.